lib/runtime/offchain/httpset.go

```diff
@@ -0,0 +1,103 @@
+package offchain
+
+import (
+	"errors"
+	"net/http"
+	"sync"
+)
+
+const maxConcurrentRequests = 1000
+
+var (
+	errIntBufferEmpty        = errors.New("int buffer exhausted")
+	errIntBufferFull         = errors.New("int buffer is full")
+	errRequestIDNotAvailable = errors.New("request id not available")
+)
+
+// requestIDBuffer created to control the amount of available non-duplicated ids
+type requestIDBuffer chan int16
+
+// newIntBuffer creates the request id buffer starting from 1 till @buffSize (by default @buffSize is 1000)
+func newIntBuffer(buffSize int16) *requestIDBuffer {
```

Comment:
does this need to return a pointer? you can change the receiver methods below to be `(b requestIDBuffer)`, then you don't need to dereference it in the methods

Reply:
Done!